ci/Jenkinsfile.test-e2e.android

```diff
@@ -0,0 +1,256 @@
+          echo "Using provided BrowserStack app id: ${env.BROWSERSTACK_APP_ID}"
+        }
+      }
+    }
+
+    stage('Download') {
+      when {
+        allOf {
+          expression { !params.BROWSERSTACK_APP_ID?.trim() }
+          expression { params.BUILD_SOURCE?.startsWith('http') }
+        }
+      }
+      steps { timeout(5) { script { dir('test/e2e_appium') {
+        if (!params.BUILD_SOURCE?.trim()) {
+          error('Specify BUILD_SOURCE when BROWSERSTACK_APP_ID is empty.')
+        }
+        sh 'mkdir -p ./pkg/'
+        fileOperations([
+          fileDownloadOperation(
+            url:            params.BUILD_SOURCE,
+            targetFileName: 'downloaded.apk',
+            targetLocation: './pkg/',
+            userName:       '',
+            password:       '',
+          )
+        ])
+        def apkPath = utils.findFile('test/e2e_appium/pkg/*.apk')
+        if (!apkPath) {
+          error("Unable to locate APK under pkg/. Ensure BUILD_SOURCE produces pkg/*.apk artifacts.")
+        }
+        env.APK_PATH = apkPath
+        echo "APK ready at ${env.APK_PATH}"
+      } } } }
+    }
+
+    stage('Copy') {
+      when {
+        allOf {
+          expression { !params.BROWSERSTACK_APP_ID?.trim() }
+          expression { !params.BUILD_SOURCE?.startsWith('http') }
+        }
+      }
+      steps { timeout(5) { script { dir('test/e2e_appium') {
+        if (!params.BUILD_SOURCE?.trim()) {
+          error('Specify BUILD_SOURCE when BROWSERSTACK_APP_ID is empty.')
+        }
+        copyArtifacts(
+          projectName: params.BUILD_SOURCE,
+          filter:      'pkg/*.apk',
+          selector:    lastWithArtifacts(),
+          target:      './'
+        )
+        def apkPath = utils.findFile('test/e2e_appium/pkg/*.apk')
+        if (!apkPath) {
+          error("Unable to locate APK under pkg/. Ensure BUILD_SOURCE produces pkg/*.apk artifacts.")
+        }
+        env.APK_PATH = apkPath
+        echo "APK ready at ${env.APK_PATH}"
+      } } } }
+    }
+
+    stage('Upload APK to BrowserStack') {
+      when {
+        expression { !env.BROWSERSTACK_APP_ID }
+      }
+      steps {
+        script {
+          withCredentials([
+            usernamePassword(
+              credentialsId: 'browserstack-status-desktop',
+              usernameVariable: 'BROWSERSTACK_USERNAME',
+              passwordVariable: 'BROWSERSTACK_ACCESS_KEY'
+            )
+          ]) {
+            def response = sh(
+              script: "./scripts/upload_browserstack_apk.sh",
+              returnStdout: true
+            ).trim()
+            def result = readJSON text: response
+            def appUrl = result?.app_url
+            if (!appUrl) {
+              error("BrowserStack upload failed: ${response}")
+            }
+            env.BROWSERSTACK_APP_ID = appUrl
+            env.BROWSERSTACK_BUILD_NAME = result?.custom_id
+            env.BROWSERSTACK_BUILD_IDENTIFIER = env.BUILD_NUMBER
```

Comment:
Where is `BUILD_NUMBER` coming from?

Reply:
I assumed Jenkins global vars are available (https://ci.status.im/pipeline-syntax/globals#env). Is that correct?